src/oidrequest.c

```diff
@@ -931,10 +931,26 @@ Return Value:
 
         break;
 
-        // TODO: Inplement these query information requests.
-    case OID_GEN_RECEIVE_BUFFER_SPACE:
-    case OID_GEN_MAXIMUM_SEND_PACKETS:
-    case OID_GEN_TRANSMIT_QUEUE_LENGTH:
+	case OID_GEN_TRANSMIT_BUFFER_SPACE:
+		ulInfo = 1UL;
+		pInfo = &ulInfo;
+		status = NDIS_STATUS_SUCCESS;
+	break;
```

Comment:
As we do not have an underlying NIC hardware,  are these some bogus values? Could you please clarify why 1 here and 2, 3 etc below, and whether these could adversely affect performance? As per the docs a protocol can query these and adjust the size of data per send.

Unless there is a rationale behind these numbers, defining a reasonable max for uncompleted sends & recvs to permit and set the buffer values and max send packets etc based on that could be an option ?

Reply:
In this latest revision, leaving them uninitialized should break things badly. If the protocol drivers don't think they can transmit, no traffic should flow.

The hard question is, on the RX and TX paths, what are the resource constraints? Can we always handle arbitrary amounts of traffic, or is there a preallocated set of buffers in use? I suspect the latter, in which case these could be MAX_UINT or (arbitrarily) 4MiB.

Unfortunately my port work did not involve these paths, so I'm also unsure what the best path is.